src/components/onboarding/wizard.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react"
+import Button from "../buttons/default"
+
+interface Props {
+  stepComponents: Array<React.ComponentClass<any>>
+}
+
+interface State {
+  currentStep: number
+  nextButtonEnabled: boolean
+}
+
+class Wizard extends React.Component<Props, State> {
+  currentStepComponent: any
+
+  constructor(props, state) {
+    super(props, state)
+
+    this.state = {
+      currentStep: 0,
+      nextButtonEnabled: false,
+    }
+  }
+
+  getCurrentStep(): JSX.Element | null {
+    const currentStep = this.state.currentStep
+    if (currentStep > this.props.stepComponents.length - 1) {
+      return null
+    }
+
+    const Step = this.props.stepComponents[currentStep]
+    return <Step onStateChange={this.onStepStateChanged.bind(this)} ref={step => (this.currentStepComponent = step)} />
+  }
```

Comment:
Are the on-boarding steps actually always going to be sequential, or will there be conditional steps like we have in Eigen? E.g. iirc we only ask the user to follow genes if they didn’t follow any artists.

Reply:
I'm not sure that's a requirement for this version but if it's something we'd like to do in the future we should take it into account when implementing `nextStep`.

@xtina-starr Do you know if we should have the ability to skip some steps?

Reply:
@alloy @l2succes the steps are sequential but similar to eigen in that if the user follows 4 or more artists they will skip the categories section. That's noted in the [specs](https://app.zeplin.io/project/59b2e6a22224266545e2aea2/screen/59b2e7a8cfddb7ce01bee3d5).